tests/test_clients_GcpClient.py

```diff
@@ -748,6 +751,47 @@ def test_download_from_blobstore_raises_exception_on_invalid_container(self):
             Exception, self.gcpClient._download_from_blobstore, 'blob', invalid_blob_path)
         self.gcpClient.container = prev_container
 
+    @patch('lib.clients.BaseClient.requests.post')
+    @patch('lib.clients.BaseClient.requests.get')
+    def test_gcp_client_creation(self,  mock_get, mock_post):
+        credhub_config = {
+            'type': 'online',
+            'backup_guid': 'backup-guid',
+            'instance_id': 'vm-id',
+            'secret': 'xyz',
+            'job_name': 'service-job-name',
+            'container': valid_container,
+            'projectId': project_id,
+            'credhub_url': '/credhub',
+            'credhub_uaa_url': '/oauth',
+            'credhub_key': '/test',
+            'credhub_client_id': '1',
+            'credhub_client_secret': 'secret',
+            'credhub_username': 'name',
+            'credhub_user_password': 'pwd'
+        }
+        auth_response = {
+            'access_token': 'auth-token-response'
+        }
+        mock_post.return_value = Mock(ok=True)
+        mock_post.return_value.json.return_value = auth_response
+        mock_get.return_value = Mock(ok=True)
+        mock_get.return_value.json.return_value = {
```

Comment:
As said above, please check if you can use side_effect instead.

Reply:
like we discussed, will let this way of mocking be as-is, as it also serves as a way to mock when patching does not help.